cmd/pod-scaler/admission.go

```diff
@@ -78,15 +84,94 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) admissio
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
 }
 
-func mutatePod(pod *corev1.Pod, build *buildv1.Build) {
+func mutatePodLabels(pod *corev1.Pod, build *buildv1.Build) {
 	if pod.Labels == nil {
 		pod.Labels = map[string]string{}
 	}
-	for _, label := range []string{steps.LabelMetadataOrg, steps.LabelMetadataRepo, steps.LabelMetadataBranch, steps.LabelMetadataVariant, steps.LabelMetadataTarget, steps.LabelMetadataStep} {
+	for _, label := range []string{steps.LabelMetadataOrg, steps.LabelMetadataRepo, steps.LabelMetadataBranch, steps.LabelMetadataVariant, steps.LabelMetadataTarget} {
 		buildValue, buildHas := build.Labels[label]
 		_, podHas := pod.Labels[label]
 		if buildHas && !podHas {
 			pod.Labels[label] = buildValue
 		}
 	}
 }
+
+// useOursIfLarger updates fields in theirs when ours are larger
+func useOursIfLarger(allOfOurs, allOfTheirs *corev1.ResourceRequirements) {
+	for _, item := range []*corev1.ResourceRequirements{allOfOurs, allOfTheirs} {
+		if item.Requests == nil {
+			item.Requests = corev1.ResourceList{}
+		}
+		if item.Limits == nil {
+			item.Limits = corev1.ResourceList{}
+		}
+	}
+	for _, pair := range []struct {
+		ours, theirs *corev1.ResourceList
+	}{
+		{ours: &allOfOurs.Requests, theirs: &allOfTheirs.Requests},
+		{ours: &allOfOurs.Limits, theirs: &allOfTheirs.Limits},
+	} {
+		for _, field := range []corev1.ResourceName{corev1.ResourceCPU, corev1.ResourceMemory} {
+			our := (*pair.ours)[field]
+			their := (*pair.theirs)[field]
+			if our.Cmp(their) == 1 {
+				(*pair.theirs)[field] = our
+			}
+		}
+	}
+}
+
+func mutatePodResources(pod *corev1.Pod, server *resourceServer) {
```

Comment:
note to self: we could handle rehearsals here by:

 - detecting with the rehearsal label
 - parsing the prowjob spec env as json
 - reading the extra_refs for the one with workdir:true
 - using those values for the meta we index into the cache with